src/test/java/net/minecraftforge/debug/ConfigTest.java

```diff
@@ -99,4 +133,48 @@ public SubCat(String value)
             }
         }
     }
+
+    public static class ConfigurationGui extends GuiConfig
```

Comment:
@LexManos I've now implemented an constructor for GuiConfig accepting the annotated classes as parameter. But due to the limmitation of IModGuiFactory.mainConfigGuiClass(), the modder still needs to create his own, but rather small, class to return in this method. I've also added documentation to the new and changed methods. Would you mind having another look at it?